gamehub-backend/src/test/java/com/example/gamehubbackend/services/ReviewServiceUnitTest.java

```diff
@@ -0,0 +1,99 @@
+package com.example.gamehubbackend.services;
+
+import com.example.gamehubbackend.exceptions.ReviewNotFoundException;
+import com.example.gamehubbackend.models.Review;
+import com.example.gamehubbackend.models.ReviewDTO;
+import com.example.gamehubbackend.repositories.ReviewRepository;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class ReviewServiceUnitTest {
+
+    private final ReviewRepository reviewRepository = mock(ReviewRepository.class);
+    private final IdService idService = mock(IdService.class);
+    private final ReviewService reviewService = new ReviewService(reviewRepository, idService);
+
+    @Test
+    void getAllReviews_Test() {
+        List<Review> reviews = List.of(
+                new Review("1", "user1", "game1", "username1", 4.5, "Great game!", "2020-01-01"),
+                new Review("2", "user2", "game2", "username2", 3.5, "Good game", "2020-01-02")
+        );
+
+        when(reviewRepository.findAll()).thenReturn(reviews);
+
+        List<Review> actualReviews = reviewService.getAllReviews();
+        assertEquals(reviews, actualReviews);
+        verify(reviewRepository).findAll();
+    }
+
+    @Test
+    void getReviewsByGameId_Test() {
+        String gameId = "game1";
+        List<Review> reviews = List.of(
+                new Review("1", "user1", gameId, "username1", 4.5, "Great game!", "2020-01-01")
+        );
+
+        when(reviewRepository.findByGameId(gameId)).thenReturn(reviews);
+
+        List<Review> actualReviews = reviewService.getReviewsByGameId(gameId);
+        assertEquals(reviews, actualReviews);
+        verify(reviewRepository).findByGameId(gameId);
+    }
+
+    @Test
+    void addReview_Test() {
+        ReviewDTO reviewDTO = new ReviewDTO("user1", "game1", "username1", 4.5, "Great game!", "2020-01-01");
+        Review reviewToSave = new Review("1", "user1", "game1", "username1", 4.5, "Great game!", "2020-01-01");
+
+        when(idService.randomId()).thenReturn("1");
+        when(reviewRepository.save(any(Review.class))).thenReturn(reviewToSave);
```

Comment:
i mean you are already specifying the reviewToSave, so why not just also use that here and with the verify?
```suggestion
        when(reviewRepository.save(reviewToSave)).thenReturn(reviewToSave);
```

Reply:
same in updateReview_Test